### ISSUE TITLE
Please upload Lua 5.2-compatible version to luarocks.org

### ISSUE THREAD

**Issue by rrthomas** [norman/telescope#21]
"luarocks install telescope" currently installs version 0.4.1, which doesn't work with Lua 5.2. Confusingly, it doesn't fail to install, even on Lua 5.2. Please could you update the version on luarocks.org?

**Comment by rrthomas** [comment]
(I see this is mentioned in the README, but the mention is well below the installation instructions, and anyway, why not just upload the new version rather than document that it's not uploaded? It's easy to release direct from github with luarocks, no need to make tarballs.)

**Comment by norman** [comment]
Sorry, this completely slipped my mind - I had intended to do this ages ago.

Can you release directly from Github without an SCM rockspec and no tarballs now? What's the state of the art now that they've done away with downloads?

**Comment by rrthomas** [comment]
I don't know what solutions others use, but I maintain a "release" branch with complete sources, and point the rockspec at that. Telescope seems to have no generated sources, so doesn't need a separate branch: merely tag the release and point the rockspec at the git repo plus the relevant tag, something like:

```
source = {
  url = "git://github.com/luaposix/luaposix.git"
  branch = "v0.5"
}
```

(I'm using GNU autotools, so need to check in the sources it generates, but which are not, as generated sources, on my master branch, so that users don't also need autotools installed.)

**Comment by norman** [comment]
Ah, ok. The only "problem" with that is that developers have to have Git installed to install via Luarocks. I suspect for 95% or more of users that's not an issue but it would probably be ideal to not make it required. Do you ever get any complaints from people? Am I being overly concerned?

**Comment by rrthomas** [comment]
I agree it looks like a concern in theory, but in practice I've been surprised to find far more resistance to using LuaRocks than to any other dependency of the build/install chain. I've also had comments about the use of GNU autotools, in particular, complaints about its complexities when trying to use it directly, which I find odd, because LuaRocks covers those up; and also complaints from Windows users, which I find frustrating, as one reason I use LuaRocks is precisely because it has good Windows support.

I don't recall a single complaint about the use of git, but I have only been doing git releases for a month or so.

**Comment by catwell** [comment]
+1 @rrthomas, there are many rockspecs which depend on git now so I basically consider it a dependency of LuaRocks.

So please release a 5.2-compatible version. :)

**Comment by norman** [comment]
Sorry for the ridiculous delay on this, I have no excuses. I've just submitted a rockspec for 0.6.0 to the list. This new version has a few bug fixes and should be used in place of 0.5.0.
